test/built-ins/RegExp/escape/not-escaped-underscore.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2024 Leo Balter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-encodeforregexescape
+description: Ensures the underscore character is not escaped
+info: |
+  RegExp.escape ( string )
+
+  This method produces a new string in which certain characters have been escaped.
+  These characters are: . * + ? ^ $ | ( ) [ ] { } \ , - = < > # & ! % : ; @ ~ ' ` " and white space or line terminators.
+features: [RegExp.escape]
+---*/
+
+assert.sameValue(RegExp.escape('_'), '_', 'Single underscore character is not escaped');
+assert.sameValue(RegExp.escape('__'), '__', 'Thunderscore character is not escaped');
+assert.sameValue(RegExp.escape('hello_world'), '\\x68ello_world', 'String starting with ASCII letter and containing underscore is correctly escaped');
```

Comment:
the first character is only escaped differently if it's a decimal digit
```suggestion
assert.sameValue(RegExp.escape('hello_world'), 'hello_world', 'String starting with ASCII letter and containing underscore is not escaped');
```

Reply:
> `1. If _escaped_ is the empty String, and _c_ is matched by |DecimalDigit| or |AsciiLetter|, then`

Reply:
oh whoops, you're right. thanks, i'll update my polyfill.